fondant/pipeline.py

```diff
@@ -0,0 +1,447 @@
+        pipeline_versions = self.client.list_pipeline_versions(pipeline_id).versions
+        return [getattr(version, "id") for version in pipeline_versions]
+
+    def delete_pipeline(self, pipeline_name: str):
+        """
+        Function that deletes the pipeline name
+        Args:
+            pipeline_name: the name of the pipeline to delete.
+        """
+        pipeline_id = self.get_pipeline_id(pipeline_name)
+        if pipeline_id is not None:
+            pipeline_version_ids = self.get_pipeline_version_ids(pipeline_id)
+            # All versions need to be first deleted
+            for pipeline_version_id in pipeline_version_ids:
+                self.client.delete_pipeline_version(pipeline_version_id)
+            self.client.delete_pipeline(pipeline_id)
+
+            logger.info(
+                f"Pipeline {pipeline_name} already exists. Deleting old pipeline..."
+            )
+        else:
+            logger.info(f"No existing pipeline under `{pipeline_name}` name was found.")
+
+    def compile_and_upload(
+        self,
+        *,
+        pipeline: FondantPipeline,
+        delete_pipeline_package: t.Optional[bool] = False,
+    ):
+        """
+        Uploads a pipeline package to Kubeflow Pipelines and deletes any existing pipeline with the
+         same name.
+
+        Args:
+            pipeline: The fondant pipeline to compile and upload
+            delete_pipeline_package: Whether to delete the pipeline package file
+             after uploading. Defaults to False.
+
+        Raises:
+            Exception: If there was an error uploading the pipeline package.
+        """
+        self.delete_pipeline(pipeline.name)
+
+        pipeline.compile_pipeline()
+
+        logger.info(f"Uploading pipeline: {pipeline.name}")
+
+        try:
+            self.client.upload_pipeline(
+                pipeline_package_path=pipeline.package_path, pipeline_name=pipeline.name
+            )
+        except Exception as e:
+            raise Exception(f"Error uploading pipeline package: {str(e)}")
+
+        # Delete the pipeline package file if specified.
+        if delete_pipeline_package:
+            Path(pipeline.package_path).unlink()
+
+    def compile_and_run(
+        self,
+        pipeline: FondantPipeline,
+        run_name: t.Optional[str] = None,
+        experiment_name: t.Optional[str] = "Default",
+    ):
+        """
+        Compiles and runs the specified pipeline.
+
+        Args:
+            pipeline: The pipeline object to be compiled and run.
+            run_name: The name of the run. If not provided, the pipeline's name with 'run' appended
+             will be used.
+            experiment_name: The name of the experiment where the pipeline will be run.
+            Default is 'Default'.
+        """
+        pipeline.compile_pipeline()
+
+        try:
+            experiment = self.client.get_experiment(experiment_name=experiment_name)
+        except ValueError:
```

Comment:
Ideally the error would be more specific

Reply:
That's the only error that KFP throws when an experiment is not created, Added a logging to make it more specific to the user that a new experiment is created